model/task/task.go

```diff
@@ -3358,12 +3361,16 @@ func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions)
 		pipeline = append(pipeline, AddBuildVariantDisplayName...)
 	}
 	if len(opts.Statuses) > 0 {
-		pipeline = append(pipeline, bson.M{
-			"$match": bson.M{
-				DisplayStatusKey: bson.M{"$in": opts.Statuses},
-			},
-		})
+		// If a user is filtering for all statuses, it is more performant to skip the status filter
+		if !utility.StringSliceContains(opts.Statuses, "all") {
+			pipeline = append(pipeline, bson.M{
+				"$match": bson.M{
+					DisplayStatusKey: bson.M{"$in": opts.Statuses},
+				},
+			})
+		}
```

Comment:
Hmm.. I'm confused because it looks like we already pass `[]` when `"All"` is selected... 🤔

<img width="1231" alt="Screen Shot 2022-02-15 at 4 16 21 PM" src="https://user-images.githubusercontent.com/47064971/154150808-7f74817b-a1bf-4a71-9018-1fec289017ca.png">

maybe it's not being passed properly? because I do see `all` being passed 

<img width="1165" alt="Screen Shot 2022-02-15 at 4 20 14 PM" src="https://user-images.githubusercontent.com/47064971/154151110-eb2295ef-2325-499c-aca0-c59b0f2c2031.png">

Reply:
That is possible Thanks for pointing it out. I will be updating that logic as well and removing that statement since now its more efficient to include "all" in this query.

Reply:
^^ Also, this is only for the Tasks page, I don't think we have this logic in the Commits page.

I also think this function might not work as expected? When clicking `"All"` I actually see every statuses _except_ `all` in the backend, when I think it should really just be an empty array.